From 6aee0a2c72ec7d8c56fe2f9ed0ded93c3b6aaa50 Mon Sep 17 00:00:00 2001
From: "Haelwenn (lanodan) Monnier" <contact@hacktivis.me>
Date: Tue, 12 Nov 2019 03:23:49 +0100
Subject: [PATCH 1/2] sysdep: fix tests under musl+LLVM

Also needs a -Werror=implicit-function-declaration as for some reason
the linker seems to accept it at configure/detection time but not
compile time.
---
 sysdep/hasutmploginlogout.h-yes.c | 3 ---
 sysdep/hasutmplogwtmp.h-yes.c     | 2 --
 2 files changed, 5 deletions(-)

diff --git a/sysdep/hasutmploginlogout.h-yes.c b/sysdep/hasutmploginlogout.h-yes.c
index f938143..f6870f9 100644
--- a/sysdep/hasutmploginlogout.h-yes.c
+++ b/sysdep/hasutmploginlogout.h-yes.c
@@ -3,9 +3,6 @@
 #include <sys/time.h>
 #include <utmp.h>
 
-extern void login(const struct utmp *);
-extern int logout(const char *);
-
 static void foo(void) {
     struct utmp ut = {0};
     login(&ut);
diff --git a/sysdep/hasutmplogwtmp.h-yes.c b/sysdep/hasutmplogwtmp.h-yes.c
index 550c5f0..87dea83 100644
--- a/sysdep/hasutmplogwtmp.h-yes.c
+++ b/sysdep/hasutmplogwtmp.h-yes.c
@@ -1,7 +1,5 @@
 /* Public domain. */
 
-extern void logwtmp(const char *, const char *, const char *);
-
 static void foo(void) {
     logwtmp("", "", "");
 }

From 8ad1a14c677d3bfd1ed2f1e50f74acfc92ed1014 Mon Sep 17 00:00:00 2001
From: "Haelwenn (lanodan) Monnier" <contact@hacktivis.me>
Date: Tue, 12 Nov 2019 04:04:42 +0100
Subject: [PATCH 2/2] sysdep: remove other extern function declarations

---
 sysdep/haslogintty.h-yes.c | 2 --
 sysdep/hasopenpty.h-yes.c  | 2 --
 2 files changed, 4 deletions(-)

diff --git a/sysdep/haslogintty.h-yes.c b/sysdep/haslogintty.h-yes.c
index f94b96a..c7f9762 100644
--- a/sysdep/haslogintty.h-yes.c
+++ b/sysdep/haslogintty.h-yes.c
@@ -1,6 +1,4 @@
 /* Public domain. */
-extern int login_tty(int);
-
 static void foo(void) {
     login_tty(0);
 }
diff --git a/sysdep/hasopenpty.h-yes.c b/sysdep/hasopenpty.h-yes.c
index fa2b807..756a4c0 100644
--- a/sysdep/hasopenpty.h-yes.c
+++ b/sysdep/hasopenpty.h-yes.c
@@ -3,8 +3,6 @@
 #include <termios.h>
 #include <sys/ioctl.h>
 
-extern int openpty(int *, int *, char *, struct termios *, struct winsize *);
-
 static void foo(void) {
     int master, slave;
     openpty(&master, &slave, 0, 0, 0);
